### PR TITLE
RF02-TC-046_JesusDavidPierolaA

### PR DESCRIPTION
**Pull request by jdpapiero**
RF02-TC-046_JesusDavidPierolaA
All changes

### CODE REVIEW

**Comment by jdpapiero** [comment]
image

image

**Comment by jdpapiero** [comment]
image

image

**Comment by jdpapiero** [comment]
image

image

**Comment by jdpapiero** [comment]
image

image

**Comment by jdpapiero** [comment]
image

image

**Comment by jdpapiero** [comment]
image

image

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 1]
```diff
@@ -296,6 +299,48 @@ public void menuPersonalSeleccionado() throws Throwable {
         menuPersonalContract = menu.selectSubMenuPersonalContract();
     }
 
+    //Structure organizational
+    @And("^seleccionar la opcion 'Estructura organizacional' en el sub menu de 'Estructura organizacional'$")
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java:303]: your menu and option have the same name?

File: src/test/resources/features/StructureOrganizational/structureOrganizational.feature [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+Feature: Estructura organizacional
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario: Verificar contenido de estructura organizacional
+    Given 'Menu principal' estee cargado
+    When seleccionar la opcion 'Estructura organizacional' en el menu principal
+    And seleccionar la opcion 'Estructura organizacional' en el sub menu de 'Estructura organizacional'
```
Comment (marbusjim) [inline src/test/resources/features/StructureOrganizational/structureOrganizational.feature:14]: your menu and option have the same name?

File: src/test/resources/features/Position/backButtonCreatePosition.feature [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+Feature: Validar boton Atras en creacion de cargos
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario: Validar boton Atras en modo creacion de Cargo
+    Given 'Menu principal' estee cargado
+    When seleccionar la opcion 'Estructura organizacional' en el menu principal
+    And seleccionar la opcion 'Cargos del personal' en el sub menu de estructura organizacional
```
Comment (marbusjim) [inline src/test/resources/features/Position/backButtonCreatePosition.feature]: please change to "seleccionar la opcion 'Cargos del personal' en el sub menu de 'estructura organizacional'"

File: src/test/resources/features/Position/sendButtonCreatePosition.feature [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+Feature: Validar boton Enviar en creacion de cargos
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario: Validar que no permita crear un Cargo sin datos obligatorios
+    Given 'Menu principal' estee cargado
+    When seleccionar la opcion 'Estructura organizacional' en el menu principal
+    And seleccionar la opcion 'Cargos del personal' en el sub menu de estructura organizacional
```
Comment (marbusjim) [inline src/test/resources/features/Position/sendButtonCreatePosition.feature]: please change to "seleccionar la opcion 'Cargos del personal' en el sub menu de 'estructura organizacional'"

File: src/test/resources/features/Position/sendButtonCreatePosition.feature [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    And seleccionar el boton 'Crear nuevo cargo' en la pagina 'Cargos de la empresa'
+    Then verificar que exista el boton 'Enviar' en la pagina 'Crear nuevo cargo'
```
Comment (marbusjim) [inline src/test/resources/features/Position/sendButtonCreatePosition.feature]: I think you need validate the form it's empty

File: src/test/resources/features/Position/validCreatePosition.feature [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+Feature: Validar componentes creacion de cargos
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario: Validar componentes del formulario de creacion de cargo
+    Given 'Menu principal' estee cargado
+    When seleccionar la opcion 'Estructura organizacional' en el menu principal
+    And seleccionar la opcion 'Cargos del personal' en el sub menu de estructura organizacional
```
Comment (marbusjim) [inline src/test/resources/features/Position/validCreatePosition.feature]: please change to "seleccionar la opcion 'Cargos del personal' en el sub menu de 'estructura organizacional'"

File: src/test/resources/features/Position/validListPosition.feature [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+Feature: Cabeceras del listado de cargos
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario: Verificar cabecera de lista de Cargos
+    Given 'Menu principal' estee cargado
+    When seleccionar la opcion 'Estructura organizacional' en el menu principal
+    And seleccionar la opcion 'Cargos del personal' en el sub menu de estructura organizacional
```
Comment (marbusjim) [inline src/test/resources/features/Position/validListPosition.feature]: please change to "seleccionar la opcion 'Cargos del personal' en el sub menu de 'estructura organizacional'"

File: src/test/resources/features/Position/validListPosition.feature [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    Then verificar que se cargue el listado de cargos en la pagina 'Cargos de la empresa'
+    And verificar que se cargue la cabecera 'Nombre cargo' en la tabla de lista de cargos
```
Comment (marbusjim) [inline src/test/resources/features/Position/validListPosition.feature]: please add quotation marks at this line "lista de cargos" in all assertion steps

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java (hunk 1 above)
Comment (jdpapiero) [reply src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java:303]: yes

File: src/test/resources/features/StructureOrganizational/structureOrganizational.feature (hunk 2 above)
Comment (jdpapiero) [reply src/test/resources/features/StructureOrganizational/structureOrganizational.feature:14]: yes

File: src/test/resources/features/Position/backButtonCreatePosition.feature (hunk 3 above)
Comment (jdpapiero) [reply src/test/resources/features/Position/backButtonCreatePosition.feature]: done

File: src/test/resources/features/Position/sendButtonCreatePosition.feature (hunk 4 above)
Comment (jdpapiero) [reply src/test/resources/features/Position/sendButtonCreatePosition.feature]: Done

File: src/test/resources/features/Position/validCreatePosition.feature (hunk 6 above)
Comment (jdpapiero) [reply src/test/resources/features/Position/validCreatePosition.feature]: Done

File: src/test/resources/features/Position/validListPosition.feature (hunk 7 above)
Comment (jdpapiero) [reply src/test/resources/features/Position/validListPosition.feature]: Done

File: src/test/resources/features/Position/validListPosition.feature (hunk 8 above)
Comment (jdpapiero) [reply src/test/resources/features/Position/validListPosition.feature]: Done

File: src/test/resources/features/Position/sendButtonCreatePosition.feature (hunk 5 above)
Comment (jdpapiero) [reply src/test/resources/features/Position/sendButtonCreatePosition.feature]: Done, I add a validation for input field if this are empty

File: src/test/resources/features/Position/validCreatePosition.feature [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario: Validar componentes del formulario de creacion de cargo
+    Given 'Menu principal' estee cargado
+    When seleccionar la opcion 'Estructura organizacional' en el menu principal
+    And seleccionar la opcion 'Cargos del personal' en el sub menu de 'Estructura organizacional'
+    And seleccionar el boton 'Crear nuevo cargo' en la pagina 'Cargos de la empresa'
+    Then verificar que exista el campo 'Nombre' en el formulario de crear cargo
```
Comment (marbusjim) [inline src/test/resources/features/Position/validCreatePosition.feature]: please add quotation mark in 'crear cargo'

File: src/test/resources/features/Position/validCreatePosition.feature [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    And verificar que exista el campo 'Nivel' en el formulario de crear cargo
```
Comment (marbusjim) [inline src/test/resources/features/Position/validCreatePosition.feature]: please add quotation mark in 'crear cargo'

File: src/test/resources/features/Position/validCreatePosition.feature [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    And verificar que exista el campo 'Descripcion' en el formulario de crear cargo
```
Comment (marbusjim) [inline src/test/resources/features/Position/validCreatePosition.feature]: please add quotation mark in 'crear cargo'

File: src/test/resources/features/Position/validCreatePosition.feature [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+    And verificar que exista la lista desplegable 'Cargo padre' en el formulario de crear cargo
```
Comment (marbusjim) [inline src/test/resources/features/Position/validCreatePosition.feature]: please add quotation mark in 'crear cargo'

**Review by ivanMise — APPROVED** [review]

**Review by marbusjim — APPROVED** [review]
Approve

**Review by danielrpg — APPROVED** [review]
its okay

**Pull request merged**